clojure/src/info/lotharschulz/armstrong_numbers/armstrong_numbers.clj

```diff
@@ -0,0 +1,54 @@
+(ns info.lotharschulz.armstrong-numbers.armstrong-numbers
+  (:import [java.Math]))
+
+(defn get-number-of-digits [^Integer n]
+  {:pre [(not (nil? n)) (integer? n)]}
+  "Returns the :getNumberOfDigits of n if n is an integer and greater than -1"
+  (if (zero? n)
+    1
+    (if (pos? n )
+      (inc (int (Math/log10 n )))
+      nil)))
+
+(defn get-digits-alt [^Integer n]
+  {:pre [(some? n) (integer? n)]}
+  "return number of digits of given int n"
+    (if (zero? n)
+      [0]
+      (if (pos? n)
+        (into [] (map #(Character/getNumericValue %) (str n)))
+        [])))
+
+(defn get-digits [^Integer n]
+  {:pre [(some? n) (integer? n)]}
+  "return number of digits of given int n"
```

Comment:
This is by the way not a documentation string, as it is placed after the argument ist: https://github.com/bbatsov/clojure-style-guide#docstring-after-fn-name